contrib/server/kafka/server.go

```diff
@@ -0,0 +1,119 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-kratos/kratos/v2/log"
+	"github.com/go-kratos/kratos/v2/transport"
+
+	"golang.org/x/sync/errgroup"
+)
+
+var _ transport.Server = (*Server)(nil)
+
+// Message is a message abstraction
+type Message interface {
+	Key() string
+	Value() []byte
+}
+
+// Handler is used to handle the received message
+type Handler interface {
+	Topic() string
+	Handle(message Message) error
+}
+
+// Consumer receives and uses the handler to process messages
+type Consumer interface {
+	Topic() string
+	RegisterHandler(handler Handler)
+	RegisterErrorHandler(handler Handler)
+	Consume(ctx context.Context) error
+	Close() error
+}
+
+// Server is a Kafka server wrapper
+type Server struct {
+	consumers     []Consumer
+	handlers      map[string]Handler
+	errorHandlers map[string]Handler
+	logger        log.Helper
+}
+
+// ServerOption is a Kafka server option.
+type ServerOption func(server *Server)
+
+// ErrorHandlers registers a set of errorHandlers to the Server.
+func ErrorHandlers(errorHandlers []Handler) ServerOption {
+	return func(server *Server) {
+		for _, handler := range errorHandlers {
+			server.errorHandlers[handler.Topic()] = handler
+		}
+	}
+}
+
+// NewServer creates a Kafka server by options.
+func NewServer(consumers []Consumer, handlers []Handler, opts ...ServerOption) (transport.Server, error) {
```

Comment:
In my opinion, handlers and consumers are closely related. User should register the handler on their own.Relating consumer and handler by topic is not precise. Sometime two consumers consume same topic from different kafka.

Reply:
Sounds reasonable! If then, the NewConsumer should be changed as 
`func NewConsumer(brokers []string, topic, group string, handler kafka.Handler, opts ...ConfigOption) (*Consumer, error) `.
And we won't be able to initialize the Server via wire because it doesn't know which handler should be used to initialize which consumer.

Reply:
I don't think whether or not we use Wire has anything to do with the framework. User still can inject dependence by Wire. For example, they can inject consumer instead of handler. It is just the matter of how to use Wire.

Reply:
https://cloud.google.com/solutions/event-driven-architecture-pubsub